daal4py/sklearn/test/test_run_to_run_stability_tests.py

```diff
@@ -41,295 +40,269 @@
 from sklearn.datasets import make_classification
 from sklearn.datasets import make_regression
 
-##################################################
+import pytest
+
+WITHOUT_FIT = [
+    'DBSCAN',
+    'PCA_daal4py_corr',
+]
+
+TO_COMPUTE = [
+    'PCA_daal4py_corr',
+]
+
+def func(X, Y, model_name, model, attribute, methods):
+    clf = model
+    if model_name not in WITHOUT_FIT:
+        clf.fit(X, Y)
+    if model_name in TO_COMPUTE:
+        clf = clf.compute(X)
+
+    res = []
+    name = []
+    
+    if model_name == 'PCA_daal4py_corr':
+        pcatrans_algo = d4p.pca_transform(nComponents=X.shape[1]//2)
+        transform = pcatrans_algo.compute(X, clf.eigenvectors, clf.dataForTransform).transformedData
+        res.append(transform)
+        name.append('transform')
+
+    for i in methods:
+        if i == 'predict':

```

Comment:
I think this can be improved by iterating over all the automate methods. (see comments below)